source/common/filter/tcp_proxy.cc

```diff
@@ -251,6 +251,8 @@ Network::FilterStatus TcpProxy::initializeUpstreamConnection() {
        &read_callbacks_->upstreamHost()->cluster().stats().bind_errors_});
   upstream_connection_->connect();
   upstream_connection_->noDelay(true);
+  request_info_.onUpstreamHostSelected(conn_info.host_description_);
+  request_info_.upstream_local_address_ = upstream_connection_->localAddress().asString();
```

Comment:
It would be nice to populate this in router also, but that is a harder change due to how the conn pool stuff works. Can you add a TODO in a similar place in the router for us to look at this later?

Reply:
I'll add a TODO